File: Python Course/Software Development Module - in Python/Week10/Bin.py
"""
Bin.py
Billy Cussen
03/10/2020
"""
#Currently working on Logic here - learning how to work with Dictionaries in Python (as opposed to switch/case in Java)

class myBin:
    def __init__(self, binColor, apartmentNumber):
        self.binColor = binColor
        self.apartmentNumber = apartmentNumber
    
    def computeCollectionDay(self):
        result = "Your Collection Day will be: "

        def getCollectionDayGreen():
            print("Hello")
            if self.apartmentNumber % 2 == 0:
                print(result + "Monday")
            else:
                print(result + "Tuesday")

        def getCollectionDayBlack():
            if self.apartmentNumber % 2 == 0:
                print(result + "Wednesday")
            else:
                print(result + "Thursday")

        def getCollectionDayBrown():
            if self.apartmentNumber % 2 == 0:
                print(result + "Friday")
            else:
                print(result + "Saturday")

        choices = {'green': getCollectionDayGreen(), 'black': getCollectionDayBlack(), 'brown': getCollectionDayBrown()}
        return choices[self.binColor]

File: Java/Software Development Module - Java/Week10/Bin.java
/*
Bin.java
Billy Cussen
20/07/2020
*/

public class Bin {

    //Data Members
    private String binColor;
    private int apartmentNumber;
    private String result;

    //Constructor and Setter
    public Bin(String binColor, int apartmentNumber){
        this.binColor = binColor;
        this.apartmentNumber = apartmentNumber;
        this.result = "Your collection day is: ";
    }

    //Compute and Getter
    public String computeAndGetDay(){
        
        switch(this.binColor.toLowerCase()){

            case "green": 
                if(this.apartmentNumber % 2 == 0){
                    this.result += "Monday.";
                } else{
                    this.result += "Tuesday.";
                }
                break;
            case "brown": 
                if(this.apartmentNumber % 2 == 0){
                    this.result += "Wednesday.";
                } else{
                    this.result += "Thursday.";
                }
                break;
            case "black": 
                if(this.apartmentNumber % 2 == 0){
                    this.result += "Friday.";
                } else{
                    this.result += "Saturday";
                }
                break;
        }

        return this.result;
    }

}